### PR TITLE
VEN-637 | Transform berth dimensions to/from string to allow decimal comma

### PR DESCRIPTION
**Pull request by TatuArvela**
## Description :sparkles:

* Transform berth dimensions to/from string to allow decimal comma
* TODO: Replace with HDS number input when it becomes available

### Closes :no_good_woman:

* [VEN-637](https://helsinkisolutionoffice.atlassian.net/browse/VEN-637): Berth dimensions won't accept decimals

### Automated tests :gear:️

* Updated snapshot

### Manual testing :construction_worker_man:

* Test that the input fields and form validation
  * allow valid values
    * `4`
    * `4.4`
    * `4,4`
    * empty
  * deny invalid values
    * `-4`
    * `4a`
    * `4..4`
    * `neljä`
    * single space

### CODE REVIEW

File: src/features/harborView/forms/berthForm/BerthForm.tsx [hunk 1]
```diff
@@ -49,9 +68,9 @@ const transformValues = (values: any): Berth => {
   return {
     ...values,
     number: parseInt(number),
-    width: parseFloat(width),
-    length: parseFloat(length),
-    depth: parseFloat(depth),
+    width: parseFloat(String(width).replace(',', '.')),
+    length: parseFloat(String(length).replace(',', '.')),
+    depth: parseFloat(String(depth).replace(',', '.')),
```
Comment (Nizar-Rahme) [inline src/features/harborView/forms/berthForm/BerthForm.tsx:73]: Does it make sense to write a utility function for this conversion?

Comment (santtul) [reply src/features/harborView/forms/berthForm/BerthForm.tsx:73]: I vote for utility function 👍

File: src/features/harborView/forms/berthForm/BerthForm.tsx [hunk 2]
```diff
@@ -21,6 +21,23 @@ interface BerthFormProps extends FormProps<Berth> {
...
+  depth?: string;
+}
+
+const isNumber = (val: string | undefined): boolean => {
+  if (val === undefined) return true; // required is a separate test
+  const numberRegex = RegExp('^[-]?\\d+([.,]\\d+)?$');
+  return numberRegex.test(val);
+};
+
+const isPositive = (val: string | undefined): boolean => {
```
Comment (santtul) [inline src/features/harborView/forms/berthForm/BerthForm.tsx:36]: Could we extract all these number operations to separate util module and write unit tests for them?

File: src/features/harborView/forms/berthForm/BerthForm.tsx [hunk 3]
```diff
@@ -21,6 +21,23 @@ interface BerthFormProps extends FormProps<Berth> {
   pierOptions: Pier[];
 }
 
+interface BerthWithStringifiedDimensions extends Omit<Berth, 'width' | 'length' | 'depth'> {
```
Comment (santtul) [inline src/features/harborView/forms/berthForm/BerthForm.tsx:24]: Would `BerthFormValues` be better name, what do you think?

**Comment by TatuArvela** [comment]
I moved the utilities to `utils`, added tests for them and did some minor changes

File: src/features/harborView/forms/berthForm/utils/utils.ts [hunk 4]
```diff
@@ -16,3 +16,22 @@ export const getBerth = (berthData: INDIVIDUAL_BERTH | undefined): Berth | undef
     pier: berthData.berth.pier.properties?.identifier,
   };
 };
+
+export const isNumber = (value: string | undefined): boolean => {
+  if (value === undefined) return true; // required is a separate test
+  const numberRegex = RegExp('^-?\\d+([.,]\\d+)?$');
```
Comment (Nizar-Rahme) [inline src/features/harborView/forms/berthForm/utils/utils.ts:22]: One thing I forgot to note. don't we need a `new` operator?

Comment (TatuArvela) [reply src/features/harborView/forms/berthForm/utils/utils.ts:22]: Yeah! I'll fix that one quick. Strange, it seems to work without it.

**Pull request merged**